frontend/src/components/Right.js

```diff
@@ -1,94 +1,118 @@
+    { title: "times_run", field: "times_run"},
+]
 
-const Right = ({ tests }) => {
 
-    console.log("Right: ")
-    // const data = tests.tests;
-    // console.log(data);
+class Right extends Component {
 
-    // const tsts = tests;
 
-    const [data, setData] = useState([])
-    const columns = [
-        { title: "app_name", field: "app"},
-        { title: "entry_date", field: "entry_date"},
-        { title: "execution_time", field: "execution_time"},
-        { title: "test", field: "test"},
-        { title: "test_id", field: "test_id"},
-        { title: "test_status", field: "test_status"},
-        { title: "test_type", field: "test_type"},
-        { title: "times_run", field: "times_run"},
-    ]
+    constructor(props){
+        console.log("Right: Constructor")
+        super(props);
+        this.localData = this.props.tests.tests;        
+    }
 
-    useEffect(() =>{
-        axios.get('/api/get-dashboard-info')
-        .then(res => {
-          console.log(res);
-          const tests = res.data.tests
-        //   const apps = res.data.apps
-          setData(tests)
-        //   this.setState({
-        //     "allApplications": apps,
-        //     "tests": tests
-        //   });
-        })
-    },[])
+    componentDidMount(){
+        console.log("Right: componentDidMount")
+    }
 
+    componentDidUpdate(prevProps){
+        console.log("Right: componentDidUpdate")
 
-    return(
-        <Box display='flex' style={rightStyle} border={1}>
-            <Grid 
-                container 
-                direction="row"
-                spacing={2}
-                justify="center"
-                alignItems="center"
-            >
-                <Grid item xs={4}>
-                    {/* Graph 1 */}
-                    <Paper>
-                        {/* {props.hello} */}
-                        Graph 1
-                    </Paper>
-                </Grid>
-                <Grid item xs={4}>
-                    {/* Graph 2 */}
-                    <Paper>
-                        Graph2
-                    </Paper>
-                </Grid>
-                <Grid item xs={4}>
-                    {/* Graph 3 */}
-                    <Paper>
-                        Graph3
-                    </Paper>
-                </Grid>
-                <Grid item xs={8}>
-                    {/* Data Table */}
-                    <MaterialTable
-                        title="Test Data"
-                        data={data}
-                        columns={columns}
-                    />
-                    {/* <CustomTable data={data}/> */}
-                </Grid>
-                <Grid item xs={4}>
-                    {/* Graph 4 */}
-                    <Paper>
-                        Graph 4
-                    </Paper>
+        if(this.localData === undefined){
+            return console.log("You're clicking too quickly, slow down")
+        }
+
+        if(this.localData.length === 0){
```

Comment:
This is okay for now, but we probably should come up with a solution to prevent infinite looping